Hazel/src/Hazel/Scene/SceneSerializer.cpp

```diff
@@ -82,70 +82,86 @@ namespace Hazel {
 	{
 	}
 
-	static void SerializeEntity(YAML::Emitter& out, Entity entity)
+	template<typename T>
+	static void SerializeComponent(YAML::Emitter& out, const T& component);
```

Comment:
I would suggest to do the same as with `Scene::OnComponentAdded`
https://github.com/TheCherno/Hazel/blob/398e41367885ef5697bb0701d0fe1bc798c783dc/Hazel/src/Hazel/Scene/Scene.cpp#L104-L108
Thus:
```suggestion
	template<typename T>
	static void SerializeComponent(YAML::Emitter& out, const T& component)
	{
		static_assert(false);
	}
```

Reply:
What is the purpose of the static assert here? If we leave the default case undefined I get this message in MSVC, which seems useful enough:

`static function 'void SerializeComponent<T>(Emitter &,const T &)' declared but not defined`

Reply:
At least now it's giving a useful error saying "error here" rather than "hey, you might have forgotten to implement this one"